src/validation.cpp

```diff
@@ -192,12 +192,11 @@ bool CheckFinalTx(const CTransaction &tx, int flags)
 {
     AssertLockHeld(cs_main);
 
-    // By convention a negative value for flags indicates that the
-    // current network-enforced consensus rules should be used. In
-    // a future soft-fork scenario that would mean checking which
-    // rules would be enforced for the next block and setting the
-    // appropriate flags. At the present time no soft-forks are
-    // scheduled, so no flags are set.
+    // By convention a negative value for flags indicates that no
+    // soft-forked consensus rules should be used. Setting appropriate
```

Comment:
I think the comment was written when BIP113 code was merged but no yet activated and scheduled for activation. It's right we don't have soft-forks scheduled now, but it's false there is no flag to set (LOCKTIME_MEDIAN_TIME_PAST)

Reply:
re: https://github.com/bitcoin/bitcoin/pull/17443#discussion_r359105468

I don't understand "it's false there is no flag to set (LOCKTIME_MEDIAN_TIME_PAST)." since this code isn't setting LOCKTIME_MEDIAN_TIME_PAST, it's setting 0...

But more broadly, I don't understand why there would be a reason to keep this code if it isn't going to serve a function in a future soft fork. Why not simplify by making the CheckFinalTx `flags` argument required instead of optional, and dropping `std::max`. and dropping this comment?

Reply:
CheckFinalTx is called with STANDARD_LOCKTIME_VERIFY_FLAGS in `PreChecks` and `removeForReorg`. Given it's only mempool I think you'e right, you can hardcode that you always want to use the most recent consensus rules but I would prefer to do that as a follow-up to avoid missing something.